agent/bench-scripts/pbench-uperf

```diff
@@ -182,17 +183,17 @@ function stop_server {
 	local server=$1
 	local server_port=$2
 	local message=$3
-	local uperf_pid=`ssh $ssh_opts $server netstat -tlnp | grep ":$server_port " | awk '{print $7}' | awk -F/ '{print $1}'`
-	if [ ! -z "$uperf_pid" ]; then
+	local uperf_pids=`ssh $ssh_opts $server ss -tlnp --vsock | grep ":$server_port " | awk '{print $7}' | grep -o -e 'pid=[0-9]+' | cut -c 5-`
```

Comment:
This pipeline seems overly complex.  At the very least, given the second `grep` command, do we still need the first one and the `awk` command?  (And, couldn't we accomplish the `grep`-and-`cut` with a single `sed` or `awk` command?)

Reply:
Here is what `netstat -tlnp` output looks like:

```
[pportant@gandalf ~]$ netstat -tlnp
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.1:44321         0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:44322           0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:44323           0.0.0.0:*               LISTEN      -
tcp        0      0 127.0.0.1:4330          0.0.0.0:*               LISTEN      -
tcp        0      0 127.0.0.1:6379          0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:5355            0.0.0.0:*               LISTEN      -
tcp        0      0 192.168.122.1:53        0.0.0.0:*               LISTEN      -
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      -
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      -
tcp6       0      0 ::1:44321               :::*                    LISTEN      -
tcp6       0      0 :::44322                :::*                    LISTEN      -
tcp6       0      0 :::44323                :::*                    LISTEN      -
tcp6       0      0 ::1:4330                :::*                    LISTEN      -
tcp6       0      0 :::5355                 :::*                    LISTEN      -
tcp6       0      0 :::22                   :::*                    LISTEN      -
tcp6       0      0 ::1:631                 :::*                    LISTEN      -
tcp6       0      0 :::3000                 :::*                    LISTEN      -
```

Here is what `ss -tlnp --vsock` looks like:

```
Netid State  Recv-Q Send-Q Local Address:Port  Peer Address:PortProcess
tcp   LISTEN 0      5          127.0.0.1:44321      0.0.0.0:*
tcp   LISTEN 0      128          0.0.0.0:44322      0.0.0.0:*
tcp   LISTEN 0      128          0.0.0.0:44323      0.0.0.0:*
tcp   LISTEN 0      5          127.0.0.1:4330       0.0.0.0:*
tcp   LISTEN 0      511        127.0.0.1:6379       0.0.0.0:*
tcp   LISTEN 0      4096         0.0.0.0:5355       0.0.0.0:*
tcp   LISTEN 0      32     192.168.122.1:53         0.0.0.0:*
tcp   LISTEN 0      4096   127.0.0.53%lo:53         0.0.0.0:*
tcp   LISTEN 0      128          0.0.0.0:22         0.0.0.0:*
tcp   LISTEN 0      128        127.0.0.1:631        0.0.0.0:*
tcp   LISTEN 0      5              [::1]:44321         [::]:*
tcp   LISTEN 0      128             [::]:44322         [::]:*
tcp   LISTEN 0      128             [::]:44323         [::]:*
tcp   LISTEN 0      5              [::1]:4330          [::]:*
tcp   LISTEN 0      4096            [::]:5355          [::]:*
tcp   LISTEN 0      128             [::]:22            [::]:*
tcp   LISTEN 0      128            [::1]:631           [::]:*
tcp   LISTEN 0      4096               *:3000             *:*
```